src/se/vidstige/jadb/managers/PackageManager.java

```diff
@@ -86,11 +90,14 @@ public void uninstall(Package name) throws IOException, JadbException {
         InputStream s = device.executeShell("pm", "uninstall", name.toString());
         String result = Stream.readAll(s, Charset.forName("UTF-8"));
         verifyOperation("uninstall", name.toString(), result);
+        s.close();
     }
 
     public void launch(Package name) throws IOException, JadbException {
-        InputStream s = device.executeShell("monkey", "-p", name.toString(), "-c", "android.intent.category.LAUNCHER", "1");
-        s.close();
+        try (InputStream stream = device.executeShell("monkey", "-p", name.toString(), "-c", "android.intent.category.LAUNCHER", "1")) {
+            Stream.flushRead(stream);
+            stream.close();
```

Comment:
Close should not be needed here as you use try-with-resource? What's wring with the `readAll` method? I feel the name is clearer.